agent.go

```diff
@@ -192,7 +192,6 @@ func (c *controller) agentSetup() error {
 	if remoteAddr != "" {
 		if err := c.agentJoin(remoteAddr); err != nil {
 			logrus.Errorf("Error in agentJoin : %v", err)
-			return nil
```

Comment:
It is not strictly needed as the other node will be able to work with just getting a join notification from memberlist. But I think it is still correct to make that fix. Will update the PR.